files/etc/php/7.3/fpm/pool.d/10-logging.conf

```diff
@@ -0,0 +1,7 @@
+; Global configuration, added via pool.d to avoid having to modify the stock
+; php-fpm.conf.
+[global]
+; Pool is configured with catch_workers_output which means all output from
+; the workers are going to come at us via the error_log.
+; Make sure to write it to a standard location which can then be tailed.
+error_log = /var/log/fpm-php.www.log
```

Comment:
Is it possible to log to stdout, so we can view the logs with `docker logs`? Or does it still do that?

Reply:
`error_log = /proc/self/fd/1` and `error_log = /proc/self/fd/2` did not seem to do anything, the issue here is that we have quite some way to go up to pid 1 that gets the stdout that docker actually sees.

TBH, can't really grasp how to do "clean" stdout logging with our current setup, so I opted to just use the current stuff (`files/etc/runit/runsvdir/default/log-tailer/run`)

Reply:
I can't claim to have a working solution either :)

But since the images build on phusion base images we should be fine if we can log to syslog, I think:

* https://github.com/phusion/baseimage-docker#system-logging
* https://stackoverflow.com/questions/22575152/sending-errors-to-syslog-in-php-fpm

Reply:
Probably, but that's a new PR someone is free to write up :)

This PR is basically fixing this line https://github.com/reload/docker-drupal-php7-fpm/pull/42/files#diff-866cbd287cb27f1e428c01af9f1bb581L411

Reply:
(that is - we had a configuration that tried to direct the php output to a log-file, but we did it the wrong way, by fixing the config the log tailer will now pick it up and log output will be visible via `docker logs`)

Reply:
Fair :)

Just got me thinking :)

Reply:
Men, hov, vinder den her ikke kampen om loggen?

https://github.com/reload/docker-drupal-php7-fpm/blob/master/files/etc/php/7.4/mods-available/drupal-recommended.ini#L12

Reply:
Nope, dén der konfigurerer selve php worker processen. Den kører med ["catch_worker_output"](https://github.com/reload/docker-drupal-php7-fpm/blob/feature/better-log-handling/files/etc/php/7.4/fpm/pool.d/www.conf#L362) hvilket betyder at alt standard out/err ryger til master-processens error_log frem for `/dev/null` og dén error_log (ikke at forveksle med php error_log) bliver så skrevet til [`/var/log/fpm-php.www.log`](https://github.com/reload/docker-drupal-php7-fpm/blob/feature/better-log-handling/files/etc/php/7.4/fpm/pool.d/10-logging.conf#L7)

Reply:
Ok, sorry for the noice :)

Reply:
No problem, it's complicated enough that I appreciated looking it trough a second time :)